build-support/bin/ci.sh

```diff
@@ -130,7 +130,9 @@ fi
 if [[ "${python_three:-false}" == "true" ]]; then
   # The 3.4 end of this constraint is necessary to jive with the travis ubuntu trusty image.
   banner "Setting interpreter constraints for 3!"
-  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython>=3.4,<4"]'
+  # TODO(John Sirois): Allow `<4` when the issues with `3.7` are fixed. See:
+  #   https://github.com/pantsbuild/pants/issues/6363
+  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython>=3.4,<3.7"]'
```

Comment:
Can we bump this to 3.5? The default for libraries now seems to be to support 3.5+, and it reduces our overhead  and will allow us to use 3.5 features like `async` once we drop Py2.

Reply:
See above:
> The 3.4 end of this constraint is necessary to jive with the travis ubuntu trusty image.

I don't think so.